### PR TITLE
NR-219651: Link to related logs from errors

### PR DESCRIPTION
**Pull request by reneelpetit**

### CODE REVIEW

File: shared/webviews/editor/App.tsx
Comment (reneelpetit) [inline shared/webviews/editor/App.tsx:1]: I'm not sure what happened here exactly -- I think maybe we've deleted App.tsx?

**Review by CalvinAllen — CHANGES_REQUESTED** [review]
See previous comments

File: shared/webviews/editor/App.tsx
[the remark this answers is not in the document]
Comment (reneelpetit) [reply shared/webviews/editor/App.tsx:1]: great, thanks!

File: shared/ui/ipc/host.protocol.ts [hunk 1]
```diff
@@ -340,11 +341,14 @@ export interface OpenEditorViewNotification {
 		name?: IdeNames;
 		browserEngine?: BrowserEngines;
 	};
+
+	params?: {};
```
[the remark this answers is not in the document]
Comment (reneelpetit) [reply shared/ui/ipc/host.protocol.ts:345]: Oh yeah, Lexa and I were talking about how maybe in the future if we need to add more things we should use a params and then put it all under that. That's why it's there. I can take it out

File: shared/ui/Stream/APMLogging/APMLogSearchPanel.tsx [hunk 2]
```diff
@@ -194,20 +204,42 @@ export const APMLogSearchPanel = (props: {
...
 
 		const sinceOptions: SelectedOption[] = [
 			defaultOption,
-			{ value: "60 MINUTES AGO", label: "60 Minutes Ago" },
-			{ value: "3 HOURS AGO", label: "3 Hours Ago" },
-			{ value: "8 HOURS AGO", label: "8 Hours Ago" },
-			{ value: "1 DAY AGO", label: "1 Day Ago" },
-			{ value: "3 DAYS AGO", label: "3 Days Ago" },
-			{ value: "7 DAYS AGO", label: "7 Days Ago" },
+			{
```
[the remark this answers is not in the document]
Comment (reneelpetit) [reply shared/ui/Stream/APMLogging/APMLogSearchPanel.tsx]: oh yeah ... that's much easier :) Thanks

**Review by CalvinAllen — APPROVED** [review]

File: shared/ui/Stream/CodeError/index.tsx [hunk 3]
```diff
@@ -1509,6 +1512,38 @@ const BaseCodeError = (props: BaseCodeErrorProps) => {
...
 	};
 
+	const renderLogsIcon = () => {
+		return (
+			<Meta>
+				<LinkForExternalUrl href="#">
+					<MetaDescription>
+						<span
+							onClick={e => {
+								e.preventDefault(), openLogs();
```
Comment (bcanzanella) [inline shared/ui/Stream/CodeError/index.tsx:1522]: should be:

```
e.preventDefault();
openLogs();
```

File: shared/ui/Stream/CodeError/index.tsx [hunk 4]
```diff
@@ -1509,6 +1512,38 @@ const BaseCodeError = (props: BaseCodeErrorProps) => {
...
+	};
+
+	const openLogs = () => {
+		HostApi.instance.notify(OpenEditorViewNotificationType, {
+			panel: "logs",
+			title: "Logs",
+			entryPoint: "code_error",
+			entityGuid: props.errorGroup?.entityGuid,
+			traceId: currentCodeErrorData.traceId,
+			ide: {},
```
Comment (bcanzanella) [inline shared/ui/Stream/CodeError/index.tsx:1544]: fill `ide` out, check out other usages of OpenEditorViewNotificationType

File: shared/util/src/protocol/agent/agent.protocol.providers.ts [hunk 5]
```diff
@@ -1371,6 +1372,8 @@ export interface ObservabilityError extends ObservabilityErrorCore {
 	appName: string;
 	remote: string;
 	occurrenceId: string;
+	traceId?: string;
+	id?: string;
```
Comment (bcanzanella) [inline shared/util/src/protocol/agent/agent.protocol.providers.ts:1376]: what's this `id` ?

**Review by bcanzanella — CHANGES_REQUESTED** [review]

File: shared/agent/src/providers/newrelic/errors/observabilityErrorsProvider.ts [hunk 6]
```diff
@@ -455,7 +458,7 @@ export class ObservabilityErrorsProvider {
 					}
 					return {
 						entityGuid: entityGuid || errorGroupResponse.entityGuid,
-						traceId: returnTraceId,
+						traceId: returnTraceId || errorTraceResponse.actor.account.nrql.results[0].traceId,
```
Comment (dsellarsnr) [inline shared/agent/src/providers/newrelic/errors/observabilityErrorsProvider.ts:461]: `|| errorTraceResult.traceId` to keep it brief

**Review by dsellarsnr — APPROVED** [review]
Just the 1 comment and weird issue posted in Slack

**Comment by reneelpetit** [comment]
moved to https://github.com/TeamCodeStream/codestream/pull/2050  due to rebase changes

**Pull request closed without merging**